src/core/raster/qgsrasterlayerrenderer.h

```diff
@@ -80,6 +80,8 @@ class CORE_EXPORT QgsRasterLayerRenderer : public QgsMapLayerRenderer
 
     QgsRasterPipe *mPipe = nullptr;
 
+    QgsRasterLayer *mLayer;
```

Comment:
hi @elpaso sorry only now had a chance to look at your PR...

Having a map layer as a member of QgsRasterLayerRenderer is something that we have to strictly avoid - the problem is that render() function runs in a worker thread and it is unsafe to use any methods from the map layer (for example, the layer may have been deleted in the meanwhile). The correct way is to fetch everything we need from the map layer in constructor of QgsRasterLayerRenderer - so there should be a new flag (e.g. mProviderAllowsPrefetch) instead of a full QgsRasterLayer pointer...

Reply:
Thank you for looking at this, I'll make a follow up.

Reply:
https://github.com/qgis/QGIS/pull/36422